apollo-router/src/warp_http_server_factory.rs

```diff
@@ -340,10 +340,11 @@ where
         })
 }
 
+// TODO: do we actually wanna track query on the info level here?
 #[tracing::instrument(skip_all,
     name = "graphql_request",
     fields(
-        query = %request.query,
+        query = %request.query.clone().unwrap_or_else(|| "-".to_string()),
```

Comment:
Why set the query to "-"? It's done like that for the operation_name to match the processing in existing TS stats reporting. I'm not sure that's correct for a default query.

Reply:
ok I gotta check what the gateway does and get back to you on this one!

Reply:
as discussed, it seems to be the expected behavior, cc @glasser if you have more insights on that 🙏

Reply:
I mean, that's a choice that is made specifically in the context of "the 'stats result key'". I don't think anything else in Apollo does "- as empty operation name". I just wanted to not have an empty "comment" in that format. Probably doesn't make sense to do it anywhere other than stats result key formatting.

Reply:
Oh and I missed that this is for query — yeah, I don't know of anything that replaces an empty query with a hyphen, so this doesn't make sense (and I might suggest that the following line lose it as well and move that logic specifically to the stats_report_key function).

Reply:
sounds good to me, I'll keep the query empty for tracing purposes, and I'll open a followup issue to remove this from the operation name here, and make sure it's added on the "stats results" end.

Opened #481